modules/glib/2.82.2/overlay/glib/BUILD.bazel

```diff
@@ -0,0 +1,309 @@
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+load("@bazel_skylib//rules:run_binary.bzl", "run_binary")
+
+PUBLIC_HDRS = [
+    "deprecated/gallocator.h",
+    "deprecated/gcache.h",
+    "deprecated/gcompletion.h",
+    "deprecated/gmain.h",
+    "deprecated/grel.h",
+    "deprecated/gthread.h",
+    "galloca.h",
+    "garray.h",
+    "gasyncqueue.h",
+    "gatomic.h",
+    "gbacktrace.h",
+    "gbase64.h",
+    "gbitlock.h",
+    "gbookmarkfile.h",
+    "gbytes.h",
+    "gcharset.h",
+    "gchecksum.h",
+    "gconvert.h",
+    "gdataset.h",
+    "gdate.h",
+    "gdatetime.h",
+    "gdir.h",
+    "genviron.h",
+    "gerror.h",
+    "gfileutils.h",
+    "ggettext.h",
+    "ghash.h",
+    "ghmac.h",
+    "ghook.h",
+    "ghostutils.h",
+    "gi18n-lib.h",
+    "gi18n.h",
+    "giochannel.h",
+    "gkeyfile.h",
+    "glib-autocleanups.h",
+    "glib-object.h",
+    "glib-typeof.h",
+    "glib-unix.h",
+    "glib.h",
+    "glist.h",
+    "gmacros.h",
+    "gmain.h",
+    "gmappedfile.h",
+    "gmarkup.h",
+    "gmem.h",
+    "gmessages.h",
+    "gnode.h",
+    "goption.h",
+    "gpathbuf.h",
+    "gpattern.h",
+    "gpoll.h",
+    "gprimes.h",
+    "gprintf.h",
+    "gqsort.h",
+    "gquark.h",
+    "gqueue.h",
+    "grand.h",
+    "grcbox.h",
+    "grefcount.h",
+    "grefstring.h",
+    "gregex.h",
+    "gscanner.h",
+    "gsequence.h",
+    "gshell.h",
+    "gslice.h",
+    "gslist.h",
+    "gspawn.h",
+    "gstdio.h",
+    "gstrfuncs.h",
+    "gstring.h",
+    "gstringchunk.h",
+    "gstrvbuilder.h",
+    "gtestutils.h",
+    "gthread.h",
+    "gthreadpool.h",
+    "gtimer.h",
+    "gtimezone.h",
+    "gtrashstack.h",
+    "gtree.h",
+    "gtypes.h",
+    "gunicode.h",
+    "guri.h",
+    "gutils.h",
+    "guuid.h",
+    "gvariant.h",
+    "gvarianttype.h",
+    "gversion.h",
+]
+
+PUBLIC_HDRS_WIN32 = [
+    "gwin32.h",
+]
+
+INTERNAL_HDRS = glob(["*private.h"]) + [
+    "gbsearcharray.h",
+    "gconstructor.h",
+    "glib_trace.h",
+    "glib-init.h",
+    "glibintl.h",
+    "gmain-internal.h",
+    "gmirroringtable.h",
+    "gprintfint.h",
+    "gscripttable.h",
+    "gtranslit-data.h",
+    "gunibreak.h",
+    "gunichartables.h",
+    "gunicomp.h",
+    "gunidecomp.h",
+    "gvalgrind.h",
+    "gvariant-core.h",
+    "gvariant-internal.h",
+    "gvariant-serialiser.h",
+    "gvarianttypeinfo.h",
+    "gwakeup.h",
+    "msvc_recommended_pragmas.h",
+    "valgrind.h",
+]
+
+ALL_HDRS = glob([
+    "*.h",
+    "deprecated/*.h",
+])
+
+filegroup(
+    name = "all_hdrs",
+    srcs = ALL_HDRS,
+    visibility = ["//:__subpackages__"],
+)
+
+UNKNOWN_HDRS = [
+    hdr
+    for hdr in ALL_HDRS
+    if all([
+        hdr not in PUBLIC_HDRS,
```

Comment:
With the number of headers in this module, it could make sense to turn these into dicts to avoid a quadratic number of comparisons.

Reply:
Done -- removed `UNKNOWN_HDRS`, since it was used by the now-removed `fail()`.

I'd originally intended this for maintenance, so as to not miss a new header when updating to a new version of glib, but it's easy enough to add back in temporarily when updating.